server/svix-server/src/worker.rs

```diff
@@ -291,6 +257,116 @@ fn bytes_to_string(bytes: bytes::Bytes) -> String {
+    .await?
+    .ok_or_else(|| Error::Generic(format!("Application doesn't exist: {}", &msg.app_id)))?;
+
+    let endpoints: Vec<CreateMessageEndpoint> = create_message_app
+        .endpoints
+        .iter()
+        .filter(|endpoint| {
+            let endpoint_match = match &queue_task {
+                QueueTask::MessageV1(task) => task.endpoint_id == endpoint.id,
+                QueueTask::MessageBatchV1(_) => true
+            };
+            return endpoint_match &&
+            // No disabled or deleted endpoints ever
+               !endpoint.disabled && !endpoint.deleted &&
+            (
+                // Manual attempt types go through regardless
+                queue_task.clone().trigger_type() == MessageAttemptTriggerType::Manual
+                || (
+                        // If an endpoint has event types and it matches ours, or has no event types
+                        endpoint
+                        .event_types_ids
+                        .as_ref()
+                        .map(|x| x.0.contains(&msg.event_type))
+                        .unwrap_or(true)
+                    &&
+                        // If an endpoint has no channels accept all messages, otherwise only if their channels overlap.
+                        // A message with no channels doesn't match an endpoint with channels.
+                        endpoint
+                        .channels
+                        .as_ref()
+                        .map(|x| !x.0.is_disjoint(msg.channels.as_ref().map(|x| &x.0).unwrap_or(&HashSet::new())))
+                        .unwrap_or(true)
+            ))})
+        .cloned()
+        .collect();
+
+    // TODO: remove this section once destinations are obsolete
+    let destinations = messagedestination::Entity::secure_find_by_msg(queue_task.clone().msg_id())
+        .all(db)
+        .await
+        .unwrap();
+    if destinations.is_empty() {
+        let destinations = endpoints
+            .iter()
+            .map(|endpoint| messagedestination::ActiveModel {
+                msg_id: Set(msg.id.clone()),
+                endp_id: Set(endpoint.id.clone()),
+                next_attempt: Set(Some(Utc::now().into())),
+                status: Set(MessageStatus::Sending),
+                ..Default::default()
+            });
+        messagedestination::Entity::insert_many(destinations)
+            .exec(db)
+            .await?;
+    }
+
+    let futures: Vec<_> = endpoints
+        .iter()
+        .map(|endpoint| {
+            dispatch(
+                cfg.clone(),
+                db,
+                queue_tx,
+                msg.payload.clone(),
+                queue_task.clone().to_msg_task(endpoint.clone().id),
+                endpoint.to_owned(),
+            )
+        })
+        .collect();
+
+    let join = future::join_all(futures).await;
+
+    let errs: Vec<_> = join.iter().filter(|x| x.is_err()).collect();
+    if !errs.is_empty() {
+        return Err(Error::Generic(format!(
```

Comment:
I think we should re-add MessageTaskV1 tasks for the failed ones. No?

Reply:
Another thing here though: I'm currently tracing all the errors, but this could be pretty messy in a scenario where a message has lots of endpoints and all or many of those dispatches failed in a way that dispatch didn't already handle (eg db connection issue, etc).

Reply:
OK, let's keep it this way, though it makes me a bit sad. I think we should recover when we can recover.